### PR TITLE
%contacts activity

### PR DESCRIPTION
**Pull request by mikolajpp**
Implements activity source and events for contact book updates.
Fixes TLON-3176.

### CODE REVIEW

File: desk/app/activity.hoon [hunk 1]
```diff
@@ -140,22 +141,39 @@
...
+  =?  old  ?=(%7 -.old)  [%8 +.old]
+  ?>  ?=(%8 -.old)
   =.  state  old
+  :: ::  insert missing volume defaults to %base
+  :: ::
+  :: =/  base-volume
+  ::   (~(gut by volume-settings.state) [%base ~] *volume-map:a)
+  :: =.  volume-settings.state
+  ::   %+  ~(put by volume-settings.state)  [%base ~]
+  ::   (~(uni by base-volume) *volume-map:a)
```
Comment (arthyn) [inline desk/app/activity.hoon:153]: did we mean to have this uncommented?

**Review by arthyn — APPROVED** [review]
not sure if this was ready yet, but @latter-bolden was asking for a review. cool to see how little you had to change the agent, LGTM!

File: desk/app/activity.hoon (hunk 1 above)
Comment (mikolajpp) [reply desk/app/activity.hoon:153]: Oh yes, I deployed it on the test ship but forgot to push the commit. Thanks!

**Comment by mikolajpp** [comment]
Note that this must go in together with https://github.com/tloncorp/landscape/pull/297

File: desk/app/channels.hoon [hunk 2]
```diff
@@ -734,6 +734,9 @@
     ::
         [%x %v2 %channels full=?(~ [%full ~])]
       ``channels-2+!>((uv-channels-2:utils v-channels ?=(^ full.pole)))
+    ::
+        [%x %v3 %channels full=?(~ [%full ~])]
+      ``channels-3+!>((uv-channels-3:utils v-channels ?=(^ full.pole)))
```
Comment (Fang-) [inline desk/app/channels.hoon:739]: Don't forget to include this mark in your PR.

**Review by Fang- — COMMENTED** [review]
So close! Almost good, thanks.

**Pull request merged**